echopype/convert/set_groups_ek80.py

```diff
@@ -303,38 +303,65 @@ def set_platform(self) -> xr.Dataset:
+        time3 = np.array(time3) if time3 is not None else [np.nan]
 
-        # Handle potential nan timestamp for time1 and time2
+        # Handle potential nan timestamp for time1, time2, and time3
         time1 = self._nan_timestamp_handler(time1)
         time2 = self._nan_timestamp_handler(time2)
+        time3 = self._nan_timestamp_handler(time3)
+
+        # Set MRU1 lat lon attributes
+        latitude_mru1_attrs = self._varattrs["platform_var_default"]["latitude"].copy()
+        latitude_mru1_attrs.update(
+            {
+                "comment": "Derived from the Simrad MRU1 Datagrams which are "
+                "a wrapper of the Kongsberg Maritime Binary Datagrams."
+            }
+        )
+        longitude_mru1_attrs = self._varattrs["platform_var_default"]["longitude"].copy()
+        longitude_mru1_attrs.update(
+            {
+                "comment": "Derived from the Simrad MRU1 Datagrams which are "
+                "a wrapper of the Kongsberg Maritime Binary Datagrams."
+            }
+        ),
 
         # Assemble variables into a dataset: variables filled with nan if do not exist
         platform_dict = {"platform_name": "", "platform_type": "", "platform_code_ICES": ""}
         ds = xr.Dataset(
             {
-                "latitude": (["time1"], lat, self._varattrs["platform_var_default"]["latitude"]),
-                "longitude": (["time1"], lon, self._varattrs["platform_var_default"]["longitude"]),
+                "latitude": (
+                    ["time1"],
+                    lat_nmea,
+                    self._varattrs["platform_var_default"]["latitude"],
+                ),
+                "longitude": (
+                    ["time1"],
+                    lon_nmea,
+                    self._varattrs["platform_var_default"]["longitude"],
+                ),
                 "sentence_type": (
                     ["time1"],
                     msg_type,
                     self._varattrs["platform_var_default"]["sentence_type"],
                 ),
                 "pitch": (
                     ["time2"],
-                    np.array(self.parser_obj.mru.get("pitch", [np.nan])),
+                    np.array(self.parser_obj.mru0.get("pitch", [np.nan])),
                     self._varattrs["platform_var_default"]["pitch"],
                 ),
                 "roll": (
                     ["time2"],
-                    np.array(self.parser_obj.mru.get("roll", [np.nan])),
+                    np.array(self.parser_obj.mru0.get("roll", [np.nan])),
                     self._varattrs["platform_var_default"]["roll"],
                 ),
                 "vertical_offset": (
                     ["time2"],
-                    np.array(self.parser_obj.mru.get("heave", [np.nan])),
+                    np.array(self.parser_obj.mru0.get("heave", [np.nan])),
                     self._varattrs["platform_var_default"]["vertical_offset"],
                 ),
```

Comment:
We may have to change these later in some way because MRU1 also contains the same set of pitch/roll/heave data. Does the lasker data still have MRU0, or does it only have MRU1? Wondering if this should be a substitution, or both should exist (`rol`, `roll_mru1`, etc).

Reply:
Yeah, I agree. I also think both should exist where `roll` is mru0 and `roll_mru`.